urlx/parse.go

```diff
@@ -2,13 +2,54 @@ package urlx
 
 import (
 	"net/url"
+	"regexp"
+	"strings"
 
 	"github.com/ory/x/logrusx"
 )
 
+// winPathRegex is a regex for [DRIVE-LETTER]:
+var winPathRegex = regexp.MustCompile("^[A-Za-z]:.*")
+
+// Parse parses rawURL into a URL structure with special handling for file:// URLs
+// File URLs with relative paths (file://../file, ../file) will be returned as a
+// url.URL object without the Scheme set to "file". This is because the file
+// scheme doesn't support relative paths. Make sure to check for
+// both "file" or "" (an empty string) in URL.Scheme if you are looking for
+// a file path.
+// Use the companion function GetURLFilePath() to get a file path suitable
+// for the current operaring system.
+func Parse(rawURL string) (*url.URL, error) {
```

Comment:
I would rename this to `ParseWindowsCompatible`

Reply:
This is actually not about parsing URLs for Windows comparability, it is about parsing non standard file URLs. The reason is that Ory uses non standard file URLs in many places throughout the tests and documentation. On windows this caused additional problems since, to get around the non standard URL usage, the code that retrieved the file path didn't use the url.URL.Path field but instead used the url.URL.String() and then just stripped the first `file://` from the beginning. 
Anyway, long story short, This Parse function is written to handle the special case of these non standard file URLs so that they can be parsed in a consistent manner.
However, to make it more simplistic, I think we can drop the UNC support by removing this code:
```
	if strings.HasPrefix(lcRawURL, "\\\\") {
		// Windows UNC path
		// We extract the hostname and create an appropriate file:// URL
		// based on the hostname and the path
		host, path := extractUNCPathParts(rawURL)
		// It is safe to replace the \ with / here because this is POSIX style path
		return url.Parse("file://" + host + strings.ReplaceAll(path, "\\", "/"))
	}
```
Therefore I think it is a better idea to keep the name of this function as Parse() since it has nothing to do with Windows specific parsing.

Reply:
Well thought out argument - thank you!